src/Hazelcast.Net/Clustering/ClusterEvents.cs

```diff
@@ -503,6 +503,10 @@ private async Task AssignClusterViewsConnectionAsync(MemberConnection connection
                 // try to subscribe, relying on the default invocation timeout,
                 // so this is not going to last forever - we know it will end
                 var correlationId = _clusterState.GetNextCorrelationId();
+                
```

Comment:
I am not sure about this test. What happens is, `await WaitForConnection(cancellationToken)` above may return `null` in case of a cancellation, in which case we should break/loop. I *think* it would be cleaned this way:

`if (connection == null) continue:`

Reply:
Sure, it leads to same end. If we continue, while look will break it. Changing it.